lib/roda-route_parser.rb

```diff
@@ -10,7 +10,7 @@ def parse(input)
     end
 
     routes = []
-    regexp = /\A\s*#\s*route(?:\[(\w+)\])?:\s+(?:([A-Z|]+)?\s+)?(\S+)\s*\z/
+    regexp = /\A\s*#\s*route(?:\[([\w|\.|-]+)\])?:\s+(?:([A-Z|]+)?\s+)?(\S+)\s*\z/
```

Comment:
I'm not sure why this would fix the problem.  The part of the regexp you are changing is the name of the route, not the path of the route.  This change would allow route comments such as:

```
# route[sd|af]: GET /{foo}
# route[sd.af]: GET /{foo}
# route[sd-af]: GET /{foo}
```

The use of brackets in the route itself is already allowed, as this route is allowed both before and after the change:

```
# route[sdaf]: GET /{foo}
```